python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py

```diff
@@ -140,47 +170,40 @@ def example_job():
               aws_session_token: None
               # Optional[str]:  The session token to use when creating the client.
     """
-    return construct_s3_client(
-        max_attempts=context.resource_config["max_attempts"],
-        region_name=context.resource_config.get("region_name"),
-        endpoint_url=context.resource_config.get("endpoint_url"),
-        use_unsigned_session=context.resource_config["use_unsigned_session"],
-        profile_name=context.resource_config.get("profile_name"),
-        use_ssl=context.resource_config.get("use_ssl"),
-        verify=context.resource_config.get("verify"),
-        aws_access_key_id=context.resource_config.get("aws_access_key_id"),
-        aws_secret_access_key=context.resource_config.get("aws_secret_access_key"),
-        aws_session_token=context.resource_config.get("aws_session_token"),
+    return S3Resource.from_resource_context(context)
+
+
+class S3FileManagerResource(S3ResourceBase[S3FileManager]):
+    s3_bucket: str = Field(description="S3 bucket to use for the file manager.")
+    s3_prefix: str = Field(
+        default="dagster", description="Prefix to use for the S3 bucket for this file manager."
     )
 
+    def create_resource(self, context: InitResourceContext) -> S3FileManager:
```

Comment:
the more I see `create_resource` the more I cringe here.

I think we should consider `create_object_for_user_code` or something which will also solve the same io_manager problem and will allow us to consolidate those two methods in the base class

Reply:
I can put up a PR for this - are we OK breaking the `create_resource` path for existing resources (e.g. users who are trying the experimental APIs out?). 

We could also do some magic to find an impl of `create_resource` and defer to that if `create_object_for_user_code` isn't overridden.

Reply:
Yes it is experimental so that we have the latitude to do exactly that.